haigha/reader.py

```diff
@@ -232,7 +232,7 @@ def read_timestamp(self):
         Will raise BufferUnderflow if there's not enough bytes in the buffer.
         Will raise struct.error if the data is malformed
         """
-        return datetime.utcfromtimestamp(self.read_longlong())
+        return datetime.utcfromtimestamp(self.read_longlong() / 1e6)
```

Comment:
@lipixun: I would expect a change like this to be accompanied by integration tests that verify parsing of timestamps from different RabbitMQ constructs (not just from the exchange plugin). We want to make sure that this change doesn't break parsing of the timestamp from the other supported sources. Also, please be sure to update the docstring(s) of the impacted function(s).

I defer further feedback to @awestendorf, who is the owner of this repo.